automod/engine/effects.go

```diff
@@ -51,6 +51,8 @@ type Effects struct {
 	RecordTakedown bool
 	// Set of Blob CIDs to takedown (eg, purge from CDN) when doing a record takedown
 	BlobTakedowns []string
+	// If "true", indicates that a rule indicates that the action causing the event should be blocked or prevented
+	RejectEvent bool
```

Comment:
I'll comment on vibe, but I don't have a change request: This makes Effects feel a little grab-bag'y, because I don't think the rest of the labels and reports stuff is valid where this bool is relevant, and vice versa.

But I don't think I have a serious complaint about that because I don't know how else I'd write it.  (Introducing a smorgasboard of types for different XEffects per XContext might be an option, but I don't think it would make our lives significantly easier.  And it would also directly align our shins with the familiar pain point of "it would be nice if golang could represent sum types".)

Reply:
yeah, that checks out and aligns with my feelings on this struct. It is a bit big and amorphous in scope.

On the other hand it is partially obscured/internal, at least from rules. and all the fields are documented. and I *really* want to avoid an explosion of variant types.